distributed/tests/test_s3.py

```diff
@@ -27,59 +27,112 @@
-
+        file_contents = s3.cat('/'.join([test_bucket_name, k])) 
         assert file_contents == data
 
-    assert (read_content_from_keys('s3://distributed-test', k, anon=True) ==
-            read_content_from_keys('distributed-test', k, anon=True))
-
+    assert (s3.cat('/'.join([test_bucket_name, k])) ==
+            s3.cat('s3://' + '/'.join([test_bucket_name, k])))
+
+
+def test_seek_delimiter(s3):
+    fn = 'test/accounts.1.json'
+    data = files[fn]
+    with s3.open('/'.join([test_bucket_name, fn])) as f:
+        seek_delimiter(f, b'}', 0)
+        assert f.tell() == 0
+        f.seek(1)
+        seek_delimiter(f, b'}', 5)
+        assert f.tell() == data.index(b'}') + 1
+        seek_delimiter(f, b'\n', 5)
+        assert f.tell() == data.index(b'\n') + 1
+
+
+def test_read_block(s3):
+    import io
+    data = files['test/accounts.1.json']
+    lines = io.BytesIO(data).readlines()
+    path = 'distributed-test/test/accounts.1.json'
+    assert s3.read_block(path, 1, 35, b'\n') == lines[1]
+    assert s3.read_block(path, 0, 30, b'\n') == lines[0]
+    assert s3.read_block(path, 0, 35, b'\n') == lines[0] + lines[1]
+    assert s3.read_block(path, 0, 5000, b'\n') == data
+    assert len(s3.read_block(path, 0, 5)) == 5
+    assert len(s3.read_block(path, 0, 5000)) == len(data)
+    assert s3.read_block(path, 5000, 5010) == b''
 
-def test_list_summary_object_with_prefix_and_delimiter():
-    keys = get_list_of_summary_objects(test_bucket_name, 'nested/nested2/',
-                                       delimiter='/', anon=True)
 
-    assert len(keys) == 2
-    assert [k.key for k in keys] == [u'nested/nested2/file1',
-                                     u'nested/nested2/file2']
+@gen_cluster(timeout=60)
+def test_read_bytes(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
 
-    keys = get_list_of_summary_objects(test_bucket_name, 'nested/', anon=True)
+    futures = read_bytes(test_bucket_name+'/test/accounts.*', lazy=False)
+    assert len(futures) >= len(files)
+    results = yield e._gather(futures)
+    assert set(results).issuperset(set(files.values()))
 
-    assert len(keys) == 4
-    assert [k.key for k in keys] == [u'nested/file1',
-                                     u'nested/file2',
-                                     u'nested/nested2/file1',
-                                     u'nested/nested2/file2']
+    yield e._shutdown()
 
 
 @gen_cluster(timeout=60)
-def test_read_bytes(s, a, b):
+def test_read_bytes_block(s, a, b):
     e = Executor((s.ip, s.port), start=False)
     yield e._start()
-
-    futures = read_bytes(test_bucket_name, prefix='test/', anon=True,
+    bs = 15
+    vals = read_bytes(test_bucket_name+'/test/account*', blocksize=bs)
+    assert len(vals) == sum([(len(v) // bs + 1) for v in files.values()])
+    futures = e.compute(vals)
+    results = yield e._gather(futures)
+    assert sum(len(r) for r in results) == sum(len(v) for v in
+               files.values())
```

Comment:
Is it reasonable to test the stronger case that `b''.join(results) == b''.join(files.values())`?